### PR TITLE
Pass room info to lobby page after creating a room

### PR DESCRIPTION
**Pull request by alankbi**
### List of Changes:

* After creating a room, pass the room info to the lobby page using location
* Change socket return data to be the RoomDto
* Replace state to be user and room info on the lobby page
* Add redirect from lobby page to join page if room hasn't been joined

### Notes:

* I removed the addUser and deleteUser methods as we might not need them anymore
* Since the join page isn't hooked up, clicking enter on that page just redirects to itself for now

### Screencast and Images:

Minor UI updates:
Lobby page

### CODE REVIEW

**Comment by alankbi** [comment]
[Connect create page with lobby page and REST endpoint](https://trello.com/c/P0C59QEU/133-connect-create-page-with-lobby-page-and-rest-endpoint)

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [inline frontend/src/views/Lobby.tsx:25]: Right now, there's a brief moment where you are on the lobby before you are redirected back. Am I correct in assuming that the next PR will take care of the actual room validation?

If that is true, then maybe you don't not need all this redirect logic, and instead you could just throw an error? Since you'll end up deleting this after merging #43, if I understand correctly.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [inline frontend/src/views/Lobby.tsx:58]: Doesn't this always redirect the user? Is that intended?

It always seems to redirect for me, even when I provide a correct room ID.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [inline frontend/src/views/Lobby.tsx:72]: What if the `user` is present but the `room` isn't? Or vice versa?

If you throw an error instead of redirecting, then you could throw an error depending on whether the `user` or `room` isn't provided; otherwise, no change is necessary.

File: src/test/java/com/rocketden/main/service/RoomServiceTests.java
Comment (zpChris) [inline src/test/java/com/rocketden/main/service/RoomServiceTests.java:176]: Should you confirm that any `UserDto` nickname that is provided as well is also checked?

**Review by zpChris — CHANGES_REQUESTED** [review]
Requesting changes because I have a few nits, but largely because I have questions on the current redirect setup you have constructed for the Lobby page.

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:42]: I agree that user is better than u; if I do this, I'll need to rename the user state variable, maybe to like currentUser? Lmk if you have any preference on the name

File: frontend/src/api/User.ts
Comment (alankbi) [reply frontend/src/api/User.ts:2]: Right now it returns UserDto objects which don't include id, although that might be a good design decision we can talk about at the next meeting?

Comment (zpChris) [reply frontend/src/api/User.ts:2]: Oh ok, if they don't include the `id` then I think this is OK as-is.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [reply frontend/src/views/Lobby.tsx:42]: Ah I see -- `currentUser` or `userParam` are both good.

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:74]: I originally tried that but then realized doing so broke the back button functionality - basically, if you are pushed to the join page, then swipe back, you end up at the lobby page again, which immediately redirects you back to the join page. Idk how big of an issue that is for us though, so I can def change it back if cons outweigh the pros.

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:25]: As for flashing, yeah I think that might be one of the drawbacks for using redirect (I'm not sure if `history.push` has the same issue or not). As for the second part, I'm not too sure - I was assuming that people who navigated to `/game/lobby` but hadn't actually joined/created a game (i.e. arrived at the lobby from a join/create page redirect) would be redirected back to the join page, however I'm totally open to changing the behavior if we all had something else in mind

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:58]: Yeah if you navigate directly to `/game/lobby` it should always redirect you to the join page unless you already reached the page from the create/join pages (the current code looks only at the user/room location state rather than the roomId query param). My thinking behind this being that users need to enter a nickname then call the PUT endpoint to actually join the room, which currently all happens from the join page. Same thing as above tho - can easily change this if we don't like this design

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:72]: Not sure, my thought was just that if any of those are missing, then the user hasn't joined the room yet and needs to re-join to be allowed back on the lobby page. However I can def look into that after discussisng/making the other changes you suggested!

File: src/test/java/com/rocketden/main/service/RoomServiceTests.java
Comment (alankbi) [reply src/test/java/com/rocketden/main/service/RoomServiceTests.java:176]: Yeah good point, just updated!

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:42]: Sounds good, went with the `currentUser` option.

File: frontend/src/views/Lobby.tsx
Comment (helenli8181) [inline frontend/src/views/Lobby.tsx:99]: This is probably better for a later PR -- should we have a pop up message or something asking the user to confirm they want to delete someone?

**Review by helenli8181 — COMMENTED** [review]
This works well for me, but I'll hold off on approving for a little bit since I'm still trying to digest how some of the logic works.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [reply frontend/src/views/Lobby.tsx:74]: Ok, that makes sense! I tried replacing this with `window.location.href = '/'`, but the issue you described occurred. I think your solution is good -- my only nit would be to put a comment _briefly_ noting why you are using this instead of the `history` hook.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [reply frontend/src/views/Lobby.tsx:25]: I think in the future, there could be a brief "loading" state which the user sees, and then is redirected to a page with the message "you need to provide a valid room ID to enter the lobby". So this is definitely a fixable issue, and I think it is totally fine for now.

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [reply frontend/src/views/Lobby.tsx:58]: No, this makes sense -- thanks for the explanation!

File: frontend/src/views/Lobby.tsx
Comment (zpChris) [reply frontend/src/views/Lobby.tsx]: Since it's the same action of redirect right now, I think this is fine. Later, I think it may make sense to redirect with a message on why they weren't connected to the lobby. If you wanted to do this in this PR, that works, but it's also fine to leave it as-is and get to that later.

**Review by zpChris — APPROVED** [review]
This PR looks great! I only left one nit on a past comment to add a brief comment on why you are using `<Redirect ... />` instead of the `history` hook.

Also, I left other comments noting that in the future, we may want to display a brief message explaining why the user was redirected (perhaps on the homepage) -- for example, it could be a brief floating message at the top that says "No user was provided." or "The provided room ID was invalid.". If you wanted to add that for this PR, you can, but no need to -- I think it already is great! :)

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [reply frontend/src/views/Lobby.tsx:99]: Yeah, right now it just does a simple `console.log` since we don't have that functionality yet. Lmk if you think I should change that to a pop-up or something tho

**Comment by alankbi** [comment]
Sounds good, I added the comment and also changed the redirect to pass an error message to the join page. All that's left is to check location for an error on the join page, but since error handling on the join page is only added in #43, I'm thinking I might implement that over there to hopefully prevent merge conflicts.

**Review by helenli8181 — APPROVED** [review]
It looks great, thanks for the explanation a couple days ago! Sorry this approval came late

**Pull request merged**
